packages/next/src/server/web/sandbox/context.ts

```diff
@@ -385,6 +402,13 @@ Learn More: https://nextjs.org/docs/messages/edge-dynamic-code-evaluation`),
               : String(input)
           validateURL(url)
           super(url, init)
+
+          const store = requestStore.getStore()
+          if (store) {
+            new Headers(init?.headers).forEach((value, key) => {
+              store.headers.set(key, value)
+            })
+          }
```

Comment:
Why are we needing to augment the store from headers in a `new Request()`, shouldn't this only be done from the initial request event in the sandbox?

Reply:
that does sound correct - should be able to remove this here then

Reply:
confirmed it was unnecessary, removed it 👍🏼